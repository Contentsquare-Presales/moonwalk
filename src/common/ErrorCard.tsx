import React, { useState, useEffect } from "react";
import styled from "styled-components/native";
import { Animated, Button, Easing } from "react-native";
import { useTheme } from "@react-navigation/native";
import Icon from "../common/Icon";

const Wrapper = styled(Animated.View)`
  margin: 40px;
  border-radius: 10px;
  padding: 20px;
`;

const Title = styled.Text`
  text-align: center;
  font-weight: bold;
  font-size: 20px;
  margin-left: 10px;
`;

const Desc = styled.Text`
  text-align: center;
  margin: 10px;
`;

const Row = styled.View`
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

interface Props {
  onRetry: () => any;
  message?: string;
}

const ErrorCard = ({ message, onRetry }: Props) => {
  const { colors } = useTheme();
  const [appearAnim] = useState(new Animated.Value(0));

  useEffect(() => {
    Animated.timing(appearAnim, {
      toValue: 1,
      duration: 600,
      easing: Easing.out(Easing.quad),
    }).start();
  }, []);

  return (
    <Wrapper
      style={{
        backgroundColor: colors.secondary,
        opacity: appearAnim,
        transform: [
          {
            translateY: appearAnim.interpolate({
              inputRange: [0, 1],
              outputRange: [80, 0],
            }),
          },
        ],
      }}
    >
      <Row>
        <Icon name="Alert" color={colors.text} />
        <Title style={{ color: colors.text }}>Error</Title>
      </Row>
      <Desc style={{ color: colors.text }}>
        {message || "Something went wrong"}
      </Desc>
      <Button title="Try again" onPress={onRetry} />
    </Wrapper>
  );
};

export default ErrorCard;
