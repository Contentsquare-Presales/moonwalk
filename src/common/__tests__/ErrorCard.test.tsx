import React from "react";
import ErrorCard from "../ErrorCard";
import renderer from "react-test-renderer";

jest.mock('../../common/Icon', () => 'Icon');

describe("ErrorCard", () => {
  it("renders correctly with a default message", () => {
    const props = {
      onRetry: jest.fn(),
    };
    const tree = renderer.create(<ErrorCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with a custom message", () => {
    const props = {
      message: "$_MESSAGE_TEXT_$",
      onRetry: jest.fn(),
    };
    const tree = renderer.create(<ErrorCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});