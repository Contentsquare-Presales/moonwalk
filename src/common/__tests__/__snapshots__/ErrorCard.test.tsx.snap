// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard renders correctly with a custom message 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 10,
      "marginBottom": 40,
      "marginLeft": 40,
      "marginRight": 40,
      "marginTop": 40,
      "opacity": 0,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "transform": Array [
        Object {
          "translateY": 80,
        },
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Icon
      color="rgb(28, 28, 30)"
      name="Alert"
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 10,
            "textAlign": "center",
          },
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      Error
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "textAlign": "center",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    $_MESSAGE_TEXT_$
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Try again
      </Text>
    </View>
  </View>
</View>
`;

exports[`ErrorCard renders correctly with a default message 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 10,
      "marginBottom": 40,
      "marginLeft": 40,
      "marginRight": 40,
      "marginTop": 40,
      "opacity": 0,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "transform": Array [
        Object {
          "translateY": 80,
        },
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Icon
      color="rgb(28, 28, 30)"
      name="Alert"
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 10,
            "textAlign": "center",
          },
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      Error
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "textAlign": "center",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    Something went wrong
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Try again
      </Text>
    </View>
  </View>
</View>
`;
